python/perf-kernels/flash-attention.py

```diff
@@ -249,7 +272,14 @@ def _attn_fwd_inner(acc, l_i, m_i, q, k_ptrs, v_ptrs, bias_ptrs, stride_kn, stri
             causal_mask = OFFS_M[:, None] >= causal_boundary[None, :]
             qk = tl.where(causal_mask, qk, float("-inf"))
         # -- compute qk ----
-        qk += tl.dot(q, k)
+
+        if INT8_GEMM:
+            qk += ((((tl.dot(q, k) * q_descale)) * k_descale) * QK_SCALE).to(tl.float32)
```

Comment:
So currently this is doing

dot output in int32 -> q/k descales and qk scale -> convert to f32.

I wonder if it makes more sense to do

dot output in int32 -> convert to fp32 -> q/k descales and qk scale.

What format / value do the scales have? I think they might be float32 values which means the multiplication is doing an implicit int32 -> f32 type convert. You can confirm quickly in the IR.

Reply:
It is doing:
dot in int32 -> arith.sitofp to fp32 -> scaling and accumulation in fp32
Because all the scaling numbers are fp32 (when they are defined as fp32 in the test)

```
        %161 = tt.load %arg35 : tensor<64x64x!tt.ptr<i8>, #blocked> loc(#loc244)
        %162 = tt.load %arg36 : tensor<64x64x!tt.ptr<i8>, #blocked1> loc(#loc245)
        %163 = triton_gpu.local_alloc %161 : (tensor<64x64xi8, #blocked>) -> !tt.memdesc<64x64xi8, #shared1, #triton_gpu.shared_memory> loc(#loc244)
        %164 = triton_gpu.local_load %163 : !tt.memdesc<64x64xi8, #shared1, #triton_gpu.shared_memory> -> tensor<64x64xi8, #triton_gpu.dot_op<{opIdx = 1, parent = #mma, kWidth = 8}>> loc(#loc244)
        %165 = tt.dot %111, %164, %cst_1 : tensor<128x64xi8, #triton_gpu.dot_op<{opIdx = 0, parent = #mma, kWidth = 8}>> * tensor<64x64xi8, #triton_gpu.dot_op<{opIdx = 1, parent = #mma, kWidth = 8}>> -> tensor<128x64xi32, #mma> loc(#loc187)
        amdgpu.instruction_sched_hint loc(#loc187)
        %166 = arith.sitofp %165 : tensor<128x64xi32, #mma> to tensor<128x64xf32, #mma> loc(#loc177)
        %167 = arith.mulf %166, %153 : tensor<128x64xf32, #mma> loc(#loc177)
        %168 = arith.mulf %167, %154 : tensor<128x64xf32, #mma> loc(#loc178)
        %169 = arith.mulf %168, %cst_5 : tensor<128x64xf32, #mma> loc(#loc188)
        %170 = arith.addf %169, %cst_6 : tensor<128x64xf32, #mma> loc(#loc189)
```
I can remove the `.to(tl.float32)` if its misleading.